docs/guide/testing.rst

```diff
@@ -0,0 +1,88 @@
+Testing your bot (pytest)
+=========================
+
+This guide shows how to test your handlers **without using the real Telegram API**.
+We will use `pytest` and `pytest-asyncio`.
+
+Installation::
+
+   pip install -U pytest pytest-asyncio
+
+Simple echo handler
+-------------------
+
+**Handler:**
+
+.. code-block:: python
+
+   # app/bot.py
+   from aiogram.types import Message
+
+   async def echo_handler(message: Message):
+       await message.answer(message.text)
+
+**Test:**
+
+.. code-block:: python
+
+   # tests/test_echo.py
+   import pytest
+   from app.bot import echo_handler
+
+   @pytest.mark.asyncio
+   async def test_echo_handler():
+       sent = []
+
+       class DummyMessage:
+           def __init__(self, text):
+               self.text = text
+           async def answer(self, text):
+               sent.append(text)
```

Comment:
It's better to use special testing tools to create mocks, e.g. MagicMock, AsyncMock.